### DISCUSSION TITLE
How do you recommend preventing key clashes in Redis between multiple apps in development?

### DISCUSSION THREAD

**Discussion by ayushn21** [sidekiq/sidekiq#5496] [Q&A]
I use `redis-namespace` in development as I work on multiple Rails apps which use sidekiq. Namespacing the keys prevents clashes. I don't do this in production for all the reasons mentioned here: https://www.mikeperham.com/2015/09/24/storing-data-with-redis/.

With Sidekiq 7 dropping support for `redis-namespace`, how would you recommend keeping the jobs from multiple apps from clashing in the local dev environment? Using different databases is one option, but this is configured in code and in a team environment, there's no guarantee everyone would have that particular database free to use.

I'm not quite sure what is the best way to proceed.

**Comment by mperham** [comment]
My first recommendation is to create a convention or mapping for your apps. If this is unreliable, you could also start your own private instance of Redis on another port besides 6379 and rely on its DBs. Something like Foreman and Rails 7's `bin/dev` can make this very easy, e.g.

https://akshaykhot.com/launching-multiple-processes-using-foreman/

**Reply by ayushn21** [reply]
Thanks :)

**Reply by barrywoolgar** [reply]
I like this idea, but does it not also require a shared convention/mapping to avoid port collision? We had a similar problem with `webpack-dev-server` and "fixed" it with a shared spreadsheet and hardcoded port numbers in each app.
